### PR TITLE
[FIX] point_of_sale: raise a warning when opening session with superuser

### PR DESCRIPTION
**Pull request by aamo-odoo**
The following error occurs when you try to open a session as the superuser.

Steps to reproduce:

- Install the ``Point_of_sale`` module
- Activate developer mode / click on debugger / Become superuser
- Point of sale / click on open Register

Traceback: 
``IndexError: list index out of range``

This error occurs when we open a session as a superuser. When we attempt to open it, at line [1] ``res.users`` is returned as empty.

[1]- https://github.com/odoo/odoo/blob/585635e8afa531b8a66295a4afe0fcdd369ed188/addons/point_of_sale/models/pos_session.py#L379

sentry-5082852453

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by aamo-odoo** [comment]
Hello @odoo/sentry_reviewers 

Please review this PR 🙂

File: addons/point_of_sale/models/pos_session.py [hunk 1]
```diff
@@ -1851,7 +1851,10 @@ def _loader_params_res_users(self):
         }
 
     def _get_pos_ui_res_users(self, params):
-        user = self.env['res.users'].search_read(**params['search_params'])[0]
+        users = self.env['res.users'].search_read(**params['search_params'])
```
Comment (adwid) [inline addons/point_of_sale/models/pos_session.py:1854]: I don't know what that method is supposed to do (didn't look), but simply in terms of implementation, it should probably be like this
```suggestion
        user = self.env['res.users'].search_read(**params['search_params'], limit=1)
```
I let you handle the rest with the pos team

**Comment by aamo-odoo** [comment]
Hey @adwid 

I have made the changes that you have suggested.

Can you have a look?

Thanks 🙂

File: addons/point_of_sale/models/pos_session.py (hunk 1 above)
Comment (adwid) [reply addons/point_of_sale/models/pos_session.py]: (You still need to `[0]` after the `if`, as `user` is supposed to be a `dict`, not a `list`)

**Comment by aamo-odoo** [comment]
Hey @adwid 

Thanks for suggestions 🙂 

Can you have a look?

**Comment by aamo-odoo** [comment]
Hey @caburj 

Can I have your review?

Thanks 🙂

**Comment by caburj** [comment]
robodoo r+

**Pull request closed without merging**